office/zathura/Portfile

```diff
@@ -45,6 +45,29 @@ depends_lib         port:desktop-file-utils \
 # blacklist compilers that don't support -std=c11
 compiler.blacklist  *gcc-4.* {clang < 300}
 
+default_variants	+doc_python39
+
+variant doc_python35 conflicts doc_python36 doc_python37 doc_python38 doc_python39 description {Install manual pages using Python 3.5} {}
+variant doc_python36 conflicts doc_python35 doc_python37 doc_python38 doc_python39 description {Install manual pages using Python 3.6} {}
+variant doc_python37 conflicts doc_python35 doc_python36 doc_python38 doc_python39 description {Install manual pages using Python 3.7} {}
+variant doc_python38 conflicts doc_python35 doc_python36 doc_python37 doc_python39 description {Install manual pages using Python 3.8} {}
+variant doc_python39 conflicts doc_python35 doc_python36 doc_python37 doc_python38 description {Install manual pages using Python 3.9} {}
+
+if {[variant_isset doc_python35]} { set python.version 35 }
+if {[variant_isset doc_python36]} { set python.version 36 }
+if {[variant_isset doc_python37]} { set python.version 37 }
+if {[variant_isset doc_python38]} { set python.version 38 }
+if {[variant_isset doc_python39]} { set python.version 39 }
+
+if {[variant_isset doc_python35] || [variant_isset doc_python36] || [variant_isset doc_python37] || [variant_isset doc_python38] || [variant_isset doc_python39]} {
+	depends_build-append	port:py${python.version}-sphinx
+
+    post-patch {
+        set python.dot_version [string range ${python.version} 0 end-1].[string index ${python.version} end]
+        reinplace "s|sphinx-build|${prefix}/Library/Frameworks/Python.framework/Versions/${python.dot_version}/bin/sphinx-build|g" ${worksrcpath}/doc/meson.build
+    }
+}
+
```

Comment:
This is overly complicated. Just have a single variant called "docs" and use py39-sphinx.

Reply:
Wouldn't that force users to needlessly install python39 and all of its associated dependencies just to compile a manpage? Even if they already have, e.g., python38 and py38-sphinx installed? The above is the approach taken also by, e.g., math/gsl.

What is really wanted is for sphinx-build (installed by sphinx_select) to be in PATH, and the particular version does not matter.

Perhaps the cleanest approach is to patch `doc/meson.build` to pass `find_program` the list of directories in which sphinx-build might be located.

Reply:
I force-pushed a commit based on what you suggested. To avoid having to update the port for each new python version, I used `port:py-sphinx` as the dependency.

Reply:
We cannot use any available python or sphinx because that means we won't have [reproducible builds](https://trac.macports.org/wiki/ReproducibleBuilds). A specific one must be chosen and used. 

If you are concerned about "unnecessary" dependencies being installed, than don't use a variant but instead make it a sub-port (zathura-doc). The documentation will get built on the builder and the end user won't have to install any python at all for it.

Reply:
Thanks for the explanation. The latest forced push takes the subport approach. Good?

Reply:
Yes.